pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -344,6 +349,37 @@ func (b *BridgePodInterface) createBridge() error {
 	return nil
 }
 
+func (b *BridgePodInterface) switchPodInterfaceWithDummy() error {
+	originalPodInterfaceName := b.podInterfaceName
+	newPodInterfaceName := fmt.Sprintf("%s-nic", originalPodInterfaceName)
+
+	// Rename pod interface to free the original name for a new dummy interface
+	err := Handler.LinkSetName(b.podNicLink, newPodInterfaceName)
```

Comment:
is this function idempotent? meaning if we get a random failure somewhere in here and the function is tried again, can it succeed regardless of what occurred the previous execution?

For example, should we be checking to see if things like the dummy nic already exists and the rename occurred before doing the rename/swap. Otherwise it looks like this might fail if it gets called twice.

Reply:
Yep. Let's verify that this is idempotent. :+1:

Reply:
It is not. Will fix it.